bigquery/snippets/snippet.go

```diff
@@ -817,6 +845,84 @@ func updateTableExpiration(client *bigquery.Client, datasetID, tableID string) e
 
 }
 
+func getView(client *bigquery.Client, datasetID, viewID string) error {
+	ctx := context.Background()
+	// [START bigquery_get_view]
+	view := client.Dataset(datasetID).Table(viewID)
+	meta, err := view.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("View %s, query: %s\n", view.FullyQualifiedName(), meta.ViewQuery)
+	// [END bigquery_get_view]
+	return nil
+}
+
+func updateView(client *bigquery.Client, datasetID, viewID string) error {
+	ctx := context.Background()
+	// [START bigquery_update_view_query]
+	view := client.Dataset(datasetID).Table(viewID)
+	meta, err := view.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+
+	newMeta := bigquery.TableMetadataToUpdate{
+		// this example updates a view into the shakespeare dataset to exclude works named after kings.
+		ViewQuery: "SELECT word, word_count, corpus, corpus_date FROM `bigquery-public-data.samples.shakespeare` WHERE corpus NOT LIKE '%king%'",
+	}
+
+	if _, err := view.Update(ctx, newMeta, meta.ETag); err != nil {
+		return err
+	}
+	// [END bigquery_update_view_query]
+	return nil
+}
+
+func updateViewDelegated(client *bigquery.Client, srcDatasetID, viewDatasetID, viewID string) error {
+	ctx := context.Background()
+	// [START bigquery_grant_view_access]
+	srcDataset := client.Dataset(srcDatasetID)
+	viewDataset := client.Dataset(viewDatasetID)
+	view := viewDataset.Table(viewID)
+
+	// First, we'll add a group to the ACL for the dataset containing the view.  This will allow users within
+	// that group to query the view, but they must have direct access to any tables referenced by the view.
+	vMeta, err := viewDataset.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	vUpdateMeta := bigquery.DatasetMetadataToUpdate{
+		Access: append(vMeta.Access, &bigquery.AccessEntry{
+			Role:       bigquery.ReaderRole,
+			EntityType: bigquery.GroupEmailEntity,
+			Entity:     "example-analyst-group@google.com",
```

Comment:
Confirming this is the correct email.

Reply:
Confirmed.  Alix set this up for the canonical example.